git_wrapper/repo.py

```diff
@@ -0,0 +1,185 @@
+
+    @repo.setter
+    def repo(self, new_repo):
+        """Allows for use of an already constructed repo
+
+            :param git.Repo new_repo: An already constructed git.Repo object to use
+        """
+        self.__repo = new_repo
+
+    @property
+    def git(self):
+        """Returns the git command for a given repo"""
+        return self.repo.git
+
+    @staticmethod
+    def clone(clone_from, clone_to):
+        """Clone a repository.
+
+           :param str clone_from: The url or path to clone the repo from
+           :param str clone_to: The local path to clone to
+           :return git.Repo: Returns the newly created repo object
+        """
+        clone_to = os.path.realpath(os.path.expanduser(clone_to))
+        logging.debug("Preparing to clone repository %s into directory %s", clone_from, clone_to)
+
+        try:
+            repo = git.repo.base.Repo.clone_from(clone_from, clone_to)
+        except git.GitCommandError as ex:
+            msg = "Error cloning repository {repo}".format(repo=clone_from)
+            raise_from(exceptions.RepoCreationException(msg), ex)
+
+        return GitRepo(repo=repo)
+
+    def destroy_and_reclone(self):
+        """Deletes the current directory and reclone the repository."""
+        # Get local path for the repo
+        local_path = self.repo.working_dir
+
+        self.logger.info("Preparing to delete and reclone repo %s", local_path)
+
+        # Get all of the remotes info
+        remotes = {}
+        for r in self.repo.remotes:
+            remotes[r.name] = r.url
+
+        self.logger.debug("Remotes for %s: %s", local_path, ' '.join(list(remotes)))
+
+        if len(remotes) == 0:
+            msg = "No remotes found for repo {repo}, cannot reclone. Aborting deletion.".format(repo=local_path)
+            raise exceptions.RepoCreationException(msg)
+
+        # Select a remote for the clone, 'origin' by default
+        if "origin" in remotes.keys():
+            default_remote = "origin"
+        else:
+            default_remote = list(remotes)[0]
+
+        self.logger.debug("Default remote for cloning set to '%s'", default_remote)
+
+        # Delete the local repo
+        self.logger.info("Deleting local repo at %s", local_path)
+        shutil.rmtree(local_path, ignore_errors=True)
+
+        # Clone it again
+        repo = self.clone(remotes[default_remote], local_path)
+        self.__setup(repo=repo.repo, path=local_path)
+
+        # Recreate the remotes
+        for name, url in remotes.items():
+            if name != default_remote:
+                try:
+                    self.logger.debug("Adding remote %s", name)
+                    r = self.repo.create_remote(name, url)
+                except git.GitCommandError as ex:
+                    msg = "Issue with recreating remote {remote}. Error: {error}".format(remote=name, error=ex)
+                    raise_from(exceptions.RemoteException(msg), ex)
+
+    def describe(self, sha):
```

Comment:
This should probably move to the GitCommit class or out to a utility function.

Reply:
Yep! I just haven't looked at it yet. I'm following the outline from issue #36 at this point, https://github.com/release-depot/git_wrapper/issues/36#issuecomment-436973211